src/client/src/components/pages/dashboard/index.jsx

```diff
@@ -15,48 +16,61 @@ class index extends Component {
     worker: "",
     employment: [],
     offending: [],
+    loading: true
   };
 
-  componentWillMount() {
-    this.props.history.push("/");
+  componentWillMount = () => {
+    this.props.history.push('/');
   }
 
   componentDidMount = async () => {
     const { dispatch } = this.props.context;
     try {
-
       // axios to get the data;
-      const { 
-        countParticipant: { count :countParticipant },
-        countCourse: { count: countCourse } ,
+      const {
+        countParticipant: { count: countParticipant },
+        countCourse: { count: countCourse },
         countWorker: { count: countWorker },
         countEmployedParticipant: { count: countEmployedParticipant },
-        countOffending: { count : countOffending},
-       } =  (await axios("/api/v2/overview")).data;
+        countOffending: { count: countOffending }
+      } = (await axios("/api/v2/overview")).data;
 
       // get average for emloyed participants
-      const employedAvg = (countEmployedParticipant*100/countParticipant).toFixed(2);
+      const employedAvg = (
+        (countEmployedParticipant * 100) /
+        countParticipant
+      ).toFixed(2);
       // get count and average for not emloyed participants
-      const countNotEmployedParticipant = (countParticipant-countEmployedParticipant)
-      const notEmployedAvg = (countNotEmployedParticipant*100/countParticipant).toFixed(2);     
+      const countNotEmployedParticipant =
+        countParticipant - countEmployedParticipant;
+      const notEmployedAvg = (
+        (countNotEmployedParticipant * 100) /
+        countParticipant
+      ).toFixed(2);
       // get average for re-offending participants
-      const offendingAvg = (countOffending*100/countParticipant).toFixed(2);
-       // get count and average for not re-offending participants
-       const countNotOffending = (countParticipant-countOffending)
-       const notOffendingAvg = (countNotOffending*100/countParticipant).toFixed(2);
+      const offendingAvg = ((countOffending * 100) / countParticipant).toFixed(
+        2
+      );
+      // get count and average for not re-offending participants
+      const countNotOffending = countParticipant - countOffending;
+      const notOffendingAvg = (
+        (countNotOffending * 100) /
+        countParticipant
+      ).toFixed(2);
```

Comment:
I know this wasn't added in this pull request but this math looks like something that could be pulled out into a small helper function, since it's repeated for notOffendingAvg, offendingAvg, notEmployedAvg and employedAvg.  That would reduce the duplication and make it easier to change in the future.  For example, if you wanted to change from `toFixed(2)` to `toFixed(1)`, you'd only have to change it inside the helper function instead of changing it in these 4 places (and forgetting one).

Reply:
I totally agree with you @tomduggan85 
maybe we should do it in the next step which gives charts for each course in the `Bounce_Back` system
It can make our work more easy and readable  :ok_hand: 
we will take your advice in our consideration :+1:

Reply:
yeah , that is cool we will do it